### DISCUSSION TITLE
VM inaccesible during backup?

### DISCUSSION THREAD

**Discussion by mortenmathiasen** [abbbi/virtnbdbackup#107] [Q&A / Support] [answered]
I have been using virtnbdbackup for remote backup of a big 2.1 TB VM. Monthly full backup and daily incremental backup. Normally, both during full backup and incremental backup the VM seems running ok and it has been accessible. Suddenly, today during daily incremental backup the VM became inaccessible and the incremental backup never finalized which it normally does within a few minutes. I had to interrupt the backup process and restart the VM. Later, I retried by doing a full backup and the same thing happened. Any idea what is going on here? Can I solve this problem or do I need another backup strategy, e.g. snapshots?

**Comment by abbbi** [comment, marked as answer]
what was the situation during freeze? Was it stuck while transferring data or did the backup freeze during the fsfreze operation?
Hard to tell where the problem is in detail, what was the backup target in this situation (local FS or NFS?) More detailed information required to analyse this problem.

**Reply by mortenmathiasen** [reply]
The backup uses ssh to copy backup from source to backup destination host, and the backup command was: 
```
virtnbdbackup -U qemu+ssh://root@hostsource/system --ssh-user root -d $vm -o $vmdest -l auto
```

Backup started at 06:20:25 and then VM started freeze and was inaccessible until 08:46:48 when the VM was destroyed and restarted. The backup log output was: 
```
[2023-02-11 06:20:25] INFO virtnbdbackup - main [MainThread]: Temporary scratch file target directory: [/var/tmp]
[2023-02-11 06:20:25] INFO virtnbdbackup - startBackupJob [MainThread]: Starting backup job.
[2023-02-11 08:46:48] WARNING client - fsFreeze [MainThread]: Guest agent is not responding: Guest agent disappeared while executing command
[2023-02-11 08:46:48] ERROR virtnbdbackup - startBackupJob [MainThread]: Failed to start backup: [Operation not supported: cannot perform disk backup for inactive domain]
```

Network and CPU ressources on both backup source and destination hosts are not exhausted during the backup. Only the VM freezes. No data transferred at all - the fsFreeze operation is the one that freezes the VM.

**Reply by abbbi** [reply]
Please check this issue, other users faced a similiar situation and it was caused by certain services within the virtual
machine / a bug in the qemu agent afaikt:

https://github.com/abbbi/virtnbdbackup/issues/102
https://github.com/abbbi/virtnbdbackup/issues/59#issuecomment-1205249686

**Reply by mortenmathiasen** [reply]
Thank you so much! You really saved my day!
Problem solved! -backup is again running smoothly.

**Reply by abbbi** [reply]
> Thank you so much! You really saved my day! Problem solved! -backup is again running smoothly.

good to hear, consider pressing the sponsor button if you find this utility useful ;)
